runtime/autoload/provider/pythonx.vim

```diff
@@ -43,15 +54,17 @@ function! s:check_version(prog, ver, skip) abort
         \ ' -c "import sys; sys.stdout.write(str(sys.version_info[0]) + '.
         \ '\".\" + str(sys.version_info[1]))"'
   let min_version = (a:ver == 2) ? '2.6' : '3.3'
-  if system(a:prog . get_version) >= min_version
+  let prog_version = system(a:prog . get_version)
+  if prog_version =~ '^' . a:ver && prog_version >= min_version
     return [1, '']
   endif
-  return [0, 'Python ' . get_version . ' interpreter is not supported.']
+  return [0, 'The Python ' . prog_version . ' interpreter ' . a:prog .
+        \ ' is not suitable for providing support for Python ' . a:ver . '.']
```

Comment:
Wording is arguably more confusing now. Suggestion:

    Python X cannot provide Python Y.